<launch>
    <!-- DEFINE PARAMETERS -->
    <!-- Vehicle -->
    <param name="L_a" type="double" value="0.125" />
    <param name="L_b" type="double" value="0.125" />
    <param name="m" type="double" value="1.98" />
    <param name="I_z" type="double" value="0.24" />
    <param name="sim_flag" type="bool" value="false" />
    <param name="feature_flag" type="bool" value="true" />
    <!-- Track parameters -->
    <param name="ds" type="double" value="0.01" />
    <param name="ey" type="double" value="0.8" />
    <param name="ey_tighten" type="double" value="1.0" />
    
    <!-- Machines -->
    <machine name="odroid" address="192.168.10.213" user="odroid" timeout="20.0" env-loader="/home/odroid/barc/env_loader_shuqi.sh"/>

    <machine name="this-pc" address="192.168.10.147" timeout="20.0" env-loader="/home/mpcubuntu/barc/env_loader_pc_shuqi.sh"/>
    
    <!-- Estimator -->
    <!--node machine="this-pc" pkg="barc" type="state_estimation_KinBkMdl_mixed.py" name="state_estimation_dynamic" output="screen"-->
    <node machine="this-pc" pkg="barc" type="state_estimator.py" name="state_estimator" output="screen">
        <!-- delay in state estuiamtor -->
        <param name="delay_df" type="double" value="0.2" />
        <param name="delay_a" type="double" value="0.0" />
        <!-- ekf properties -->
        <param name="Q_x" type="double" value="0.0" />
        <param name="Q_y" type="double" value="0.0" />
        <param name="Q_vx" type="double" value="0.0" />
        <param name="Q_vy" type="double" value="0.0" />
        <param name="Q_ax" type="double" value="1.0" />
        <param name="Q_ay" type="double" value="1.0" />
        <param name="Q_psi" type="double" value="0.0" />
        <param name="Q_psiDot" type="double" value="1.0" />
        <param name="R_x" type="double" value="10.0" />
        <param name="R_y" type="double" value="10.0" />
        <param name="R_vx" type="double" value="0.1" />
        <param name="R_vy" type="double" value="0.01" />
        <param name="R_ax" type="double" value="10.0" />
        <param name="R_ay" type="double" value="10.0" />
        <param name="R_psiDot" type="double" value="0.1" />
        <!-- ukf properties -->
    </node>

    <!-- MPC CONTROL -->
    <node machine="this-pc" pkg="barc" type="LMPC_node_feature_data_collecting.jl" name="MPC" output="screen" /> 


    <!-- LOW LEVEL CONTROLLER -->
    <node machine="this-pc" pkg="barc" type="controller_low_level.py" name="low_level_controller" />

    <!-- VISUALIZATION -->
    <node machine="this-pc" pkg="barc" type="view_car_trajectory.py" name="visualization" />

    <!-- IMU -->
    <node machine="odroid" pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
    </node>

    <!-- GPS -->
    <node machine="odroid" pkg="marvelmind_nav" type="hedge_rcv_bin" name="indoor_gps" args="/dev/ttyACM1"/>
    
    <!-- ARDUINO -->
    <!--  * encoders and ultrasound sensors -->
    <node machine="odroid" pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

</launch>
